src/amd/compiler/aco_instruction_selection_setup.cpp

```diff
@@ -891,6 +891,26 @@ static bool needs_view_index_sgpr(isel_context *ctx)
    return false;
 }
 
+static inline bool
+add_fs_arg(struct isel_context *ctx, arg_info *args, unsigned &vgpr_idx, fs_input input, unsigned value, bool enablenext)
+{
+   if (ctx->fs_vgpr_args[input]) {
+      RegClass rc = input == fs_input::persp_pull_model ? v3 : v1;
```

Comment:
I'd suppose to give the RegClass as input and increment vgpr_idx += regclass.size().
When enable_next is true, again. This way you don't need to handle pull_model as special case.

Reply:
Agreed on the increment by `rc.size()`. I thought about passing the `RegClass` but it would just be a lot of repetition so I think it's fine to figure it out here.

Reply:
the function is inlined, so this gets optimized away anyway. But you could also change it to

`add_fs_arg(struct isel_context *ctx, arg_info *args, unsigned &vgpr_idx, fs_input input, unsigned value, bool enable_next, RegClass rc = v1)`

and only add the rc where it's different.

Reply:
Yep, that's the way to go.